pages/api/id/[form]/submission/report.ts

```diff
@@ -0,0 +1,253 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { logMessage } from "@lib/logger";
+import { middleware, cors, jsonValidator, sessionExists } from "@lib/middleware";
+import submissionNamesArraySchema from "@lib/middleware/schemas/submission-name-array.schema.json";
+import {
+  BatchGetCommand,
+  DynamoDBDocumentClient,
+  TransactWriteCommand,
+} from "@aws-sdk/lib-dynamodb";
+import { MiddlewareProps, WithRequired } from "@lib/types";
+import { connectToDynamo } from "@lib/integration/dynamodbConnector";
+import { NotifyClient } from "notifications-node-client";
+import { checkOne } from "@lib/cache/flags";
+
+class FailedToSendEmailThroughGCNotify extends Error {}
+
+const handler = async (req: NextApiRequest, res: NextApiResponse, props: MiddlewareProps) => {
+  // Is this feature / endpoint active
+  const vaultActive = await checkOne("vault");
+  if (!vaultActive) return res.status(404).json({ error: "Vault not active" });
+
+  const { session } = props as WithRequired<MiddlewareProps, "session">;
+
+  const userEmail = session.user.email;
+  if (userEmail === null)
+    throw new Error(
+      `User does not have an associated email address: ${JSON.stringify(session.user)} `
+    );
+
+  const formId = req.query.form;
+
+  if (Array.isArray(formId) || !formId) return res.status(400).json({ error: "Bad request" });
+
+  const submissionNames = req.body as string[];
+
+  try {
+    const dynamoDbClient = connectToDynamo();
+
+    const submissionsFromSubmissionNames = await getSubmissionsFromSubmissionNames(
+      formId,
+      submissionNames,
+      dynamoDbClient
+    );
+
+    if (submissionsFromSubmissionNames.submissionsToReport.length > 0) {
+      await report(formId, submissionsFromSubmissionNames.submissionsToReport, dynamoDbClient);
+      await notifySupport(
+        formId,
+        submissionsFromSubmissionNames.submissionsToReport.map((submission) => submission.name),
+        userEmail
+      );
+    }
+
+    logMessage.info(
+      `user:${userEmail} reported a problem with form submissions [${submissionsFromSubmissionNames.submissionsToReport.map(
+        (submission) => submission.name
+      )}] for form ID:${formId} at:${Date.now()}`
+    );
+
+    return res.status(200).json({
+      ...(submissionsFromSubmissionNames.submissionsToReport.length > 0 && {
+        reportedSubmissions: submissionsFromSubmissionNames.submissionsToReport.map(
```

Comment:
I feel like the response JSON attributes I have chosen are not the best so let me know if you have better suggestions

```
reportedSubmissions: [],
submissionNamesAlreadyReported: [],
invalidSubmissionNames: []
```

Reply:
IMO these work also :)